src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -547,6 +547,9 @@ public void Commit(IReleaseSpec releaseSpec, IWorldStateTracer stateTracer, bool
                 {
                     FlushToTree();
                 }
+
+                codeFlushTask.GetAwaiter().GetResult();
+
```

Comment:
This is explicit optimization by @benaadams not to do it here, but in the background, then in `Commit` you have wait for the one from previous block.

Reply:
Really? Then what is with the other wait at the end of the same function?

Reply:
So @benaadams is this PR acceptable or makes your solution unachievable and you are confident you will revisit it?